controllers/defaultremediation_controller_test.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"context"
+	"github.com/medik8s/default-remediation-operator/api/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("Default Remediation CR", func() {
+	Context("Defaults", func() {
+		var underTest *v1alpha1.DefaultRemediation
+
+		BeforeEach(func() {
+			underTest = &v1alpha1.DefaultRemediation{
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "default"},
+			}
+			err := k8sClient.Create(context.Background(), underTest)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(context.Background(), underTest)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("creating a resource", func() {
+			It("CR is namespace scoped", func() {
+				Expect(underTest.Namespace).To(Not(BeEmpty()))
+			})
+
+		})
+	})
+
+	Context("Reconciliation", func() {
+		var (
+			underTest      *v1alpha1.DefaultRemediation
+			objects        []runtime.Object
+			reconciler     DefaultRemediationReconciler
+			reconcileError error
+			getNHCError    error
+		)
+
+		JustBeforeEach(func() {
```

Comment:
I'm a bit confused to be honest. Usually operator-sdk is generating code in `suite_test.go` which sets up the reconciler etc., and in the tests you just use the `k8sClient` for creating and getting resources, and cretaed CRs are reconciled automatically. What version of operator-sdk did you use? Maybe an old one? 🤔

Reply:
I see your point.
Basically so far I followed the pattern I've seen in `mao` & `nhc` in which a fake client is initiated and used instead of the `k8sClient`
I'm not sure which way is better - but from what I can see the fake client has the advantage of injecting elements without actually invoking the `Create` method (i.e. `cl := fake.NewClientBuilder().WithRuntimeObjects(objects...).Build()` )
Hopefully I'm not making to much of a mess of who I understand it ... :smile: 
Let me know what you think

Reply:
MAO was created when the new env test style tests weren't a thing yet. And NHC probably copied from MAO.
If we don't stop copying, and don't use the new generated(!) test code, the new and IMHO better method will never be used.
And I think it's better because it at least uses a real etcd db, which is more like a real cluster than the fake client.

Reply:
oh, and the env test based tests can easily be reused for real e2e tests. Even easier than I thought and did for node label operator, you just need to set USE_EXISTING_CLUSTER to true :) Killer feature IMHO

Reply:
make sense, much appreciated.
I'll make some effort to use the `k8sClient`

Reply:
Well, looks `k8sClient` [limitation](https://github.com/medik8s/machine-deletion-remediation/pull/1#discussion_r634512425) is a deal breaker in this case.
When using the fake client I can pass it my made up resources, I thought that the workaround for `k8sClient` would be to create them - but it seems like it's an issue.